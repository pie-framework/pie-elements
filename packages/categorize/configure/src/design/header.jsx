import React from 'react';
import PropTypes from 'prop-types';
import Typography from '@material-ui/core/Typography';
import { withStyles } from '@material-ui/core/styles';
import classNames from 'classnames';
import { AddButton } from './buttons';

export class Header extends React.Component {
  static propTypes = {
    classes: PropTypes.object.isRequired,
    className: PropTypes.string,
    buttonLabel: PropTypes.string,
    onAdd: PropTypes.func.isRequired,
    label: PropTypes.string.isRequired,
    info: PropTypes.any
  };

  static defaultProps = {};
  render() {
    const { classes, className, onAdd, label, buttonLabel, info } = this.props;
    return (
      <div className={classNames(classes.header, className)}>
        <div className={classes.titleContainer}>
        <Typography variant="title" className={classes.title}>
          {label}
        </Typography>
        {info}
        </div>
        <AddButton onClick={onAdd} label={buttonLabel} />
      </div>
    );
  }
}
const styles = theme => ({
  header: {
    display: 'flex',
    justifyContent: 'space-between',
    marginBottom: 2 * theme.spacing.unit
  },
  titleContainer: {
    display: 'flex',
  }
});
export default withStyles(styles)(Header);
