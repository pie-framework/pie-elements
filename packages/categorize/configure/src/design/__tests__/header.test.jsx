import { shallow } from 'enzyme';
import React from 'react';

import { Header } from '../header';

describe('Header', () => {
  let w;
  let onAdd = jest.fn();
  const wrapper = (extras) => {
    const defaults = {
      classes: {
        header: 'header',
      },
      label: 'Header',
      className: 'className',
      onAdd,
    };
    const props = { ...defaults, ...extras };
    return shallow(<Header {...props} />);
  };
  describe('snapshot', () => {
    it('renders', () => {
      w = wrapper();
      expect(w).toMatchSnapshot();
    });
  });
});
