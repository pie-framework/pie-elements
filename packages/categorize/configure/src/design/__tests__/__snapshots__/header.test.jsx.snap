// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header snapshot renders 1`] = `
<div
  className="header className"
>
  <div>
    <WithStyles(Typography)
      variant="title"
    >
      Header
    </WithStyles(Typography)>
  </div>
  <WithStyles(Tooltip)
    classes={
      Object {
        "tooltip": undefined,
      }
    }
    title=""
  >
    <span>
      <WithStyles(RawAddButton)
        onClick={[MockFunction]}
      />
    </span>
  </WithStyles(Tooltip)>
</div>
`;
